testing/upload_validation.ts

```diff
@@ -933,138 +942,122 @@ const genericObjectSchema: z.ZodTypeAny = z.lazy(() =>
     .superRefine((data, context) => {
       const schema = data as GenericObjectSchema;
 
-      const validateTitleProperty = () => {
-        if (schema.titleProperty) {
-          if (!(schema.titleProperty in schema.properties)) {
+      function validateProperty(
+        propertyKey: keyof typeof schema,
+        isValidSchema: (schema: Schema & ObjectSchemaProperty) => boolean,
+        invalidSchemaMessage: string,
+      ) {
+        if (schema[propertyKey]) {
+          const propertyValue =
+            typeof schema[propertyKey] === 'string'
+              ? (schema[propertyKey] as string)
+              : (schema[propertyKey] as ObjectProperty).value;
+
+          let propertySchema =
+            typeof schema[propertyKey] === 'string' && propertyValue in schema.properties
+              ? schema.properties[propertyValue]
+              : undefined;
+          if (!propertySchema) {
+            const schemaPropertyPath = normalizePropertyValuePathIntoSchemaPath(propertyValue);
+            propertySchema = JSONPath({
+              path: schemaPropertyPath,
+              json: schema.properties,
+            })?.[0];
+          }
+
+          if (!propertySchema) {
             context.addIssue({
               code: z.ZodIssueCode.custom,
-              path: ['titleProperty'],
-              message: `The "titleProperty" field name "${schema.titleProperty}" does not exist in the "properties" object.`,
+              path: [propertyKey],
+              message: `The "${propertyKey}" field name "${propertyValue}" does not exist in the "properties" object.`,
```

Comment:
Nit, we probably want to update these messages when we interpreted the value as a path to indicate that it was a path. 'The "titleProperty" path did not resolve property in the schema.' or something.

Reply:
ah good call